src/ff1.rs

```diff
@@ -112,14 +112,14 @@ pub trait NumeralString: Sized {
 }
 
 #[derive(Clone)]
-struct Prf<CIPH: NewBlockCipher + BlockCipher> {
+struct Prf<CIPH: BlockCipher + BlockEncrypt + BlockDecrypt> {
```

Comment:
Both `BlockEncrypt` and `BlockDecrypt` are bound on `BlockCipher`, so we don't need to explicitly bound on it:
```suggestion
struct Prf<CIPH: BlockEncrypt + BlockDecrypt> {
```